### ISSUE TITLE
Changed entityDef structure did not upgrade properly for event based custom entity causing HTTP 500 + SQL ERROR 42S22

### ISSUE THREAD

**Issue by tiwj** [espocrm/espocrm#2735]
**Describe the bug**
After upgrading to the latest version a custom created entity of base type Event may no longer work properly, which could be observed this way:

- HTTP 500 when opening details
- HTTP 500 when deleting entry
- side panels and bottom panels not loaded in detail form 

Logfile contains the error:
`[2023-04-28 12:31:15] ERROR: (42S22) SQLSTATE[42S22]: Column not found: 1054 Unknown column '<ENTITY_NAME>.date_start' in 'field list'; GET /<ENTITY_NAME>/<ENTITY_ID>; line: 76, file: /var/www/html/espo/application/Espo/ORM/Executor/DefaultSqlExecutor.php
`
where <ENTITY_NAME/ID> replaces the respective entity/id.

I also posted on forum and already found the solution:
https://forum.espocrm.com/forum/installation-and-upgrade-help/91690-error-42s22-sqlstate-42s22-but-column-exists

The problem is the `duration` field in entityDef JSON. It seems that in older version, there is a valid expression like:
```
"duration": {
   ...
   "select": "TIMESTAMPDIFF(SECOUND, <ENTITY_NAME>.date_start, <ENTITY_NAME>.date_end)",
   ...
}
```
Notice the <ENTITY_NAME> here, which is also part of the error log.

In a fresh created Event based entity, the duration field looks like:
```
"duration": {
   ...
   "select": {
      "select": "TIMESTAMPDIFF(SECOUND, date_start, date_end)"
   }
   ...
}
```

Notice the entity name is missing and the structure has changed.

I assume the way this works was modified through the last versions and removing the <ENTITY_NAME> here and updating the structure was not part of the update or did not succeed.

After removing the <ENTITY_NAME> and rebuilding, everything is back to normal. We had this issue with two different entities. 

~~For the first entity, the additional database fields date_start_date and date_end_date also did not exist and the date_start/date_end field type had to be updated to DateTimeOptional.~~

~~The second entity already had DateTimeOptional with the respective DB fields, but also missed the updated duration expressions.~~

**To Reproduce**
1. Hard to reproduce: you need to have a version with event based entities which have the old entityDefs for ~~start/end date +~~ duration at version max 7.2.6 (this is where my upgrade path started causing the errors)
2. Upgrade to latest version
3. Hopefully see the expected error 500 and not updated entityDefs

I know it is hard to reproduce but I hope you know what the problem is when reading through the description and forum post.
Before the first upgrade, I also switched from PHP 7.4 to 8.2 (upgrade reported some errors on first attempt regarding PHP, so I needed to upgrade PHP first and then restart command upgrade).

**Expected behavior**
The Event based entity defs should have been migrated to latest style with ~~DateTimeOptional and~~ updated duration select/order expression

**Screenshots**
See the forum post mention above

**EspoCRM version**
Upgrade path started with [7.2.6](https://github.com/espocrm/espocrm/releases/tag/7.2.6)
To the latest version.
Error must have occured somewhere inbetween. 

*Edit: misunderstanding with DateTimeOptional is indeed optional and not part of the issue > removed*

**Comment by yurikuzn** [comment]
Affects Event entity types created before v6.0 which contain more than 1 word in their name.

**Comment by yurikuzn** [comment]
It was not supposed to migrate existing custom event entities to DateTimeOptional. It will change the behavior of existing custom entities. If one would like to have this new feature, they either can modify entityDefs or create a new entity.

**Comment by tiwj** [comment]
Okay, did not know that DateTimeOptional is indeed optional, so this was only the result of comparing old and new definitions. For me and the rest I think it is fine that you identified the source of the issue with entities containing more then one word. I can confirm both corrupted entities had more then one word and I can also cofirm another custom event based entity still works after upgrading.

Thank you for the fast reply and stay tuned!

**Comment by yurikuzn** [comment]
Added fix to 7.4 upgrade script: https://github.com/espocrm/espocrm/commit/bb59cda13175a1702fbbd23c1308e0bc60f80fd2

Thanks for reporting.

**Comment by yurikuzn** [comment]
Fix will be in the next upgrade. But if your events working, fix is not needed.
